go/kbfs/libhttpserver/content_type_utils.go

```diff
@@ -57,6 +57,73 @@ func getDisposition(defaultInlineValue bool, mimeType string) string {
 // way to do that as even Go's DetectContentType just uses utf-8 by default.
 const textPlainUtf8 = "text/plain; charset=utf-8"
 
+func beforeSemicolon(str string) string {
+	semicolonIndex := strings.Index(str, ";")
+	if semicolonIndex > 0 {
+		str = str[:semicolonIndex]
+	}
+	return strings.ToLower(strings.TrimSpace(str))
+}
+
+var supportedImgMimeTypes = map[string]bool{
+	"image/png":  true,
+	"image/jpeg": true,
+	"image/gif":  true,
+	"image/webp": true,
+}
+
+func getGetGUIFileContextResp(viewType keybase1.GUIViewType) (
+	keybase1.GUIViewType, string) {
+	return viewType, strconv.Itoa(int(viewType))
+}
+
+func getGUIFileContext(contentTypeRaw, contentDispositionRaw string) (
+	viewType keybase1.GUIViewType, invariance string) {
+	contentTypeProcessed := beforeSemicolon(contentTypeRaw)
+	disposition := beforeSemicolon(contentDispositionRaw)
+	if disposition == "attachment" {
+		return getGetGUIFileContextResp(keybase1.GUIViewType_DEFAULT)
+	}
+	switch {
+	case contentTypeProcessed == "text/plain":
+		return getGetGUIFileContextResp(keybase1.GUIViewType_TEXT)
```

Comment:
This might be more clear if you have a param outside the switch, and then make one call after the switch, like:

```go
viewType := keybase1.GUIViewType_DEFAULT
switch {
case contentTypeProcessed == "text/plain":
    viewType = keybase1.GUIViewType_TEXT
case supportedImgMimeTypes[contentTypeProcessed]:
	viewType = keybase1.GUIViewType_IMAGE
...
}
return viewType, getGetGUIFileContextResp(viewType)
```

This would avoid having `getGetGUIFileContextResp` needlessly return its input parameter.  You could probably even get rid of `getGetGUIFileContextResp()` entirely in that case.